src/api/java/CMakeLists.txt

```diff
@@ -20,9 +29,35 @@ add_custom_target(
     "${PYTHON_EXECUTABLE}"
     "${CMAKE_CURRENT_LIST_DIR}/genkinds.py"
     --kinds-header "${PROJECT_SOURCE_DIR}/src/api/cvc4cppkind.h"
-    --kinds-file-prefix "${CMAKE_CURRENT_BINARY_DIR}/cvc/Kind"
+    --kinds-file-prefix "${CMAKE_CURRENT_BINARY_DIR}/src/main/java/cvc/Kind"
   DEPENDS
     genkinds.py
   COMMENT
     "Generate Kind.java"
 )
+
+
+find_package(Java)
+include(UseJava)
+find_package(JNI REQUIRED)
+
+# specify the output destination of cvcJavaApi library
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/src/main/resources")
+
+add_library(cvcJavaApi SHARED jni/cvc_Solver.cpp)
+
+target_include_directories(cvcJavaApi PUBLIC ${JNI_INCLUDE_DIRS})
+target_include_directories(cvcJavaApi PUBLIC "${CMAKE_INSTALL_PREFIX}/include")
```

Comment:
Why exactly is this needed?

Reply:
Because of this [line](https://github.com/CVC4/CVC4/pull/6196/files#diff-31a7cb3b95442b3150f94f788c11f51614a5a4577d97e29f003b4ca2fc481af4R3)

Reply:
Yes, but for that you actually need to install the CVC4 headers to a given prefix. Why not add use the headers from the source tree with `target_include_directories(cvcJavaApi PUBLIC "${PROJECT_SOURCE_DIR}/src")`?

Reply:
Changed to use `src/api`